app/controllers/admin/territories/invitations_devise_controller.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+class Admin::Territories::InvitationsDeviseController < Devise::InvitationsController
+  def new
+    @services = Service.all
+    self.resource = resource_class.new(territories: [current_territory])
+    #  authorize(resource)
+    render :new, layout: "application_configuration"
+  end
+
+  def create
+    # invite_resource creates the new Agent in DB and sends the invitation.
+    if (agent = invite_resource)
+      authorize(Agent.new(invite_params))
+      flash[:notice] = if agent.invitation_accepted?
+                         I18n.t "activerecord.notice.models.agent_role.existing", email: agent.email
+                       else
+                         I18n.t "activerecord.notice.models.agent_role.invited", email: agent.email
+                       end
+      redirect_to admin_territory_agents_path(current_territory)
+    else
+      # Keep the error message, but redirect instead of just rendering the template:
+      # we want a new empty form.
+      flash[:error] = agent.errors.full_messages.to_sentence
+      redirect_to action: :new
+    end
+  end
+
+  def pundit_user
+    AgentTerritorialContext.new(current_agent, current_territory)
+  end
+
+  def current_territory
+    @current_territory ||= Territory.find(params[:territory_id])
+  end
+  helper_method :current_territory
+
+  def policy_scope(*args, **kwargs)
+    super([:configuration, *args], **kwargs)
+  end
+  helper_method :policy_scope
+
+  def authorize(*args, **kwargs)
+    super([:configuration, *args], **kwargs)
+  end
+
+  # invite_params is called by devise::invitationscontroller#invite_resource
+  def invite_params
+    params = devise_parameter_sanitizer.sanitize(:invite)
```

Comment:
Si on fait 
```suggestion
    params = super
```
on indique plus clairement que le but de la méthode est simplement de modifier le Hash, et ça évite qu'on se demande ce que fait la méthode `devise_parameter_sanitizer`. :wink: 

Je me demande aussi si il ne peut pas y avoir des effets de bord à modifier directement le Hash retourné par `super`, qui est en réalité extrait de `params`. N'est-il pas plus sûr d'utiliser un merge qui ne fait pas de mutation ? 

```ruby
  def invite_params
    super.merge(
      # Only ever invite to the current territory
      agent_territorial_access_rights_attributes: { "0": { territory: current_territory } },

      # the omniauth uid _is_ the email, always. note: this may be better suited in a hook in agent.rb
      uid: params[:email]
    )
  end
```

C'est juste des propositions, si tu penses que c'est pas important y'a pas de souci !

Reply:
Je trouve ça plutôt élégant comme proposition. Je vais mettre ça. Merci !